File: src/main/java/de/swerik/euler/_00/Problem0015.java
package de.swerik.euler._00;

import java.math.BigInteger;

public class Problem0015 {
    public Problem0015() {
        long startTime = System.nanoTime();
        long result = 0;

        // Solution:
        BigInteger n = BigInteger.ONE;
        BigInteger k = BigInteger.ONE;
        for (int i = 21; i < 40; i++) {
            n = n.multiply(new BigInteger(i + ""));
            k = k.multiply(new BigInteger(i - 20 + ""));
        }
        n = (n.divide(k)).multiply(new BigInteger("2"));

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + n + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}

File: src/main/cs/00/Problem0015.cs
using System;
using System.Diagnostics;
using System.Numerics;
using Euler.test.cs;

namespace Euler.main.cs._00
{
    public class Problem0015
    {
        public Problem0015()
        {
            var stopWatch = new Stopwatch();
            stopWatch.Start();

            // Solution:
            BigInteger n = 1;
            BigInteger k = 1;
            for (var i = 21; i < 40; i++)
            {
                n = n * i;
                k = k * (i - 20);
            }

            n = n / k * 2;


            stopWatch.Stop();
            var elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + n + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                   ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                   : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                     " ms"));
            if (ProblemTest.DoBenchmark)
                ProblemBenchmark.AddTime(
                    15, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
        }
    }
}